egui/src/containers/panel.rs

```diff
@@ -168,67 +166,84 @@ impl SidePanel {
+                        <= ui.style().interaction.resize_grab_radius_side;
 
-                if ctx.input().pointer.any_pressed()
-                    && ctx.input().pointer.any_down()
+                if ui.input().pointer.any_pressed()
+                    && ui.input().pointer.any_down()
                     && mouse_over_resize_line
                 {
-                    ctx.memory().interaction.drag_id = Some(resize_id);
+                    ui.memory().interaction.drag_id = Some(resize_id);
                 }
-                is_resizing = ctx.memory().interaction.drag_id == Some(resize_id);
+                is_resizing = ui.memory().interaction.drag_id == Some(resize_id);
                 if is_resizing {
                     let width = (pointer.x - side.side_x(panel_rect)).abs();
                     let width = clamp_to_range(width, width_range).at_most(available_rect.width());
                     side.set_rect_width(&mut panel_rect, width);
                 }
 
                 let dragging_something_else =
-                    ctx.input().pointer.any_down() || ctx.input().pointer.any_pressed();
+                    ui.input().pointer.any_down() || ui.input().pointer.any_pressed();
                 resize_hover = mouse_over_resize_line && !dragging_something_else;
 
                 if resize_hover || is_resizing {
-                    ctx.output().cursor_icon = CursorIcon::ResizeHorizontal;
+                    ui.output().cursor_icon = CursorIcon::ResizeHorizontal;
                 }
             }
         }
 
-        let clip_rect = ctx.input().screen_rect();
-        let mut panel_ui = Ui::new(ctx.clone(), layer_id, id, panel_rect, clip_rect);
-
-        let frame = frame.unwrap_or_else(|| Frame::side_top_panel(&ctx.style()));
+        let mut panel_ui = ui.child_ui(panel_rect, Layout::top_down(Align::Min));
+        panel_ui.expand_to_include_rect(panel_rect);
+        let frame = frame.unwrap_or_else(|| Frame::side_top_panel(ui.style()));
         let inner_response = frame.show(&mut panel_ui, |ui| {
             ui.set_min_height(ui.max_rect_finite().height()); // Make sure the frame fills the full height
             add_contents(ui)
         });
 
         let rect = inner_response.response.rect;
-        ctx.memory().id_data.insert(id, PanelState { rect });
+        ui.memory().id_data.insert(id, PanelState { rect });
 
         if resize_hover || is_resizing {
             let stroke = if is_resizing {
-                ctx.style().visuals.widgets.active.bg_stroke
+                ui.style().visuals.widgets.active.bg_stroke
             } else {
-                ctx.style().visuals.widgets.hovered.bg_stroke
+                ui.style().visuals.widgets.hovered.bg_stroke
             };
             // draw on top of ALL panels so that the resize line won't be covered by subsequent panels
-            let resize_layer = LayerId::new(Order::PanelResizeLine, Id::new("panel_resize"));
```

Comment:
This makes the resize line go on top of everything else, which is unfortunate but inevitable. Ideally we should have some way of saying "This layer goes directly on top of this other layer", i.e. let each window have it's own foreground layer. But alas, we don't have that yet.